tests/FSharp.Compiler.ComponentTests/Conformance/PatternMatching/And/andPattern04.fs

```diff
@@ -0,0 +1,21 @@
+// #Conformance #PatternMatching #PatternMatchingGuards 
+
+open System
+
+let x: Result<unit, exn> = Error (NullReferenceException())
+
+match x with
+| Error (_: exn & :? NullReferenceException) -> printfn "NullRef"
+| _ -> ()
```

Comment:
How does this fare for something like the following with an active pattern after the `&`, which [used to be allowed](https://sharplab.io/#v2:DYLgZgzgPg9gDgUwHYAIDKBPCAXBBbAWAChjgFsUAKAfRRBQQA9UAyFAMQEMBLYAVwBOCFLSgiAlCgC8DZlXHFiTAMYI42bjFQBZDAFFmiomQo06s1il0HUY6pJlNUlcUA==):

```fsharp
let (_ : exn & Failure _ | _) = exn ()
```

?

Or for a match on an exception type declared with the `exception` keyword, which was also allowed before:

```fsharp
exception MyExn

let (_ : exn & MyExn | _) = exn ()
```

?